src/modm/driver/encoder/encoder.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_ENCODER_HPP
+#define MODM_ENCODER_HPP
+
+#include <modm/platform.hpp>
+#include <type_traits>
+
+namespace modm
+{
+/**
+ * \brief	Encoder
+ *
+ * \ingroup	modm_ui_encoder
+ * \author	2021 Thomas Sommer
+ */
+struct encoder
+{
+	typedef enum Division
+	{
+		ONE,
+		TWO,
+		FOUR
+	} Division_t;
+};
+
```

Comment:
`/// @ingroup modm_driver_encoder` to make the Doxygen work right.

Reply:
The doxygen groups are automatically generated by the `modm:docs` module and they replicate the module structure, so `modm:driver:encoder` becomes `@defgroup modm_driver_encoder` `@addtogroup modm_driver` automatically.
Example: `modm:driver:encoder.output` https://docs.modm.io/develop/api/stm32f103zgt7/group__modm__driver__encoder__output.html